packages/utils/src/utils/debounce.utils.ts

```diff
@@ -9,6 +9,7 @@
  * @param {number} [timeout=300] - The debounce delay in milliseconds. Defaults to 300ms if not provided or invalid.
  * @returns {(args: unknown[]) => void} A debounced version of the original function.
  */
+// eslint-disable-next-line @typescript-eslint/no-unsafe-function-type
 export const debounce = (func: Function, timeout?: number) => {
```

Comment:
What about typing it like
```suggestion
export const debounce = (func:(...args: any[]) => unknown, timeout?: number) => {
```

or we could create a util type 
```
type CbFunction = (...args: any[]) => unknown
``` 

Alternatively, we can rely on inferring generics:
```suggestion
export const debounce = <T extends (...args: any[]) => unknown>(func: T, timeout?: number) => {
```

Reply:
Then we need an eslint rule to allow `any` which is disallowed. One or the other...
But yeah fundamentally not against something better than `Function`.
Maybe you can provide that in another?